ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/privacy/eea/EeaGetTransactionCountTest.java

```diff
@@ -0,0 +1,118 @@
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.parameters.JsonRpcParameter;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcErrorResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import tech.pegasys.pantheon.ethereum.privacy.PrivateTransactionHandler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class EeaGetTransactionCountTest {
+  final Address address = Address.fromHexString("55");
+  private Enclave enclave = mock(Enclave.class);
+  private PrivateTransactionHandler privateTransactionHandler =
+      mock(PrivateTransactionHandler.class);
+
+  private JsonRpcRequest request;
+
+  @Before
+  public void setup() {
+    final String[] privateFor = new String[] {"first", "second", "third"};
+    final String privateFrom = "thePrivateFromKey";
+
+    final Map<String, Object> jsonMap = new HashMap<>();
+    jsonMap.put("address", address.toString());
+    jsonMap.put("privateFrom", privateFrom);
+    jsonMap.put("privateFor", privateFor);
+
+    request = new JsonRpcRequest("2.0", "eea_getTransactionCount", new Object[] {jsonMap});
+  }
+
+  @Test
+  public void validRequestProducesExpectedNonce() throws Exception {
+    PrivacyGroup[] returnedGroups =
+        new PrivacyGroup[] {
+          new PrivacyGroup("Group1", Type.LEGACY, "Group1_Name", "Group1_Desc", new String[0]),
+        };
+
+    when(enclave.findPrivacyGroup(any())).thenReturn(returnedGroups);
+    when(privateTransactionHandler.getSenderNonce(address, "Group1")).thenReturn(8L);
+
+    final EeaGetTransactionCount method =
+        new EeaGetTransactionCount(new JsonRpcParameter(), enclave, privateTransactionHandler);
+
+    final JsonRpcResponse response = method.response(request);
+    assertThat(response).isInstanceOf(JsonRpcSuccessResponse.class);
+
+    final JsonRpcSuccessResponse successResponse = (JsonRpcSuccessResponse) response;
+    assertThat(successResponse.getResult()).isEqualTo("0x8");
+  }
+
+  @Test
+  public void moreThanOneMatchingLegacyGroupReturnsError() throws Exception {
+    PrivacyGroup[] returnedGroups =
+        new PrivacyGroup[] {
+          new PrivacyGroup("Group1", Type.LEGACY, "Group1_Name", "Group1_Desc", new String[0]),
+          new PrivacyGroup("Group2", Type.LEGACY, "Group2_Name", "Group2_Desc", new String[0]),
+        };
+
+    when(enclave.findPrivacyGroup(any())).thenReturn(returnedGroups);
+    when(privateTransactionHandler.getSenderNonce(address, "Group1")).thenReturn(8L);
+
+    final EeaGetTransactionCount method =
+        new EeaGetTransactionCount(new JsonRpcParameter(), enclave, privateTransactionHandler);
+
+    final JsonRpcResponse response = method.response(request);
+    assertThat(response).isInstanceOf(JsonRpcErrorResponse.class);
```

Comment:
Are you expecting a specific error message here?

Reply:
Done.